src/iso15118/d20/state/dc_charge_loop.cpp

```diff
@@ -255,6 +281,10 @@ FsmSimpleState::HandleEventReturnType DC_ChargeLoop::handle_event(AllocatorType&
             present_current = control_data->current;
         } else if (const auto control_data = ctx.get_control_event<StopCharging>()) {
             stop = *control_data;
+        } else if (const auto control_data = ctx.get_control_event<UpdateDynamicModeParameters>()) {
+            new_departure_time = *control_data->departure_time;
+            new_target_soc = *control_data->target_soc;
+            new_min_soc = *control_data->min_soc;
```

Comment:
Unsure about this, but one may express pointer semantics by using `const auto* control_data ...`

Reply:
Added `*`